### PR TITLE
Halo2 SHPLONK benchmark

### PR DESCRIPTION
**Pull request by tyshko-rostyslav**

### CODE REVIEW

File: halo2-shplonk/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+    let params = gen_srs(23);
+    let agg_circuit = AggregationCircuit::<SHPLONK>::new(&params, snarks);
+    println!("after agg_circuit");
+    let start0 = start_timer!(|| "gen vk & pk");
+    let pk = gen_pk(
+        &params,
+        &agg_circuit.without_witnesses(),
+        None,
+        // Some(Path::new("./examples/agg.pk")),
```
Comment (jonesmarvin8) [inline halo2-shplonk/src/main.rs:327]: Remove this line?

File: halo2-shplonk/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+            let partial_sbox = meta.advice_column();
+
+            let rc_a = (0..WIDTH).map(|_| meta.fixed_column()).collect::<Vec<_>>();
+            let rc_b = (0..WIDTH).map(|_| meta.fixed_column()).collect::<Vec<_>>();
+
+            meta.enable_constant(rc_b[0]);
+
+            Self::Config {
+                input: state[..RATE].try_into().unwrap(),
+                // expected,
```
Comment (jonesmarvin8) [inline halo2-shplonk/src/main.rs:86]: // expected,
 this line is not necessary.

File: halo2-shplonk/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+        pub instance: Fr,
+        pub message_arr: [Fr; L],
+        pub message: Value<[Fr; L]>,
+        pub _spec: PhantomData<S>,
+    }
+
+    #[derive(Debug, Clone)]
+    pub struct MyConfig<const WIDTH: usize, const RATE: usize, const L: usize> {
+        input: [Column<Advice>; L],
+        // expected: Column<Instance>,
```
Comment (jonesmarvin8) [inline halo2-shplonk/src/main.rs:51]: Not necessary; commented line can be removed.

**Comment by jonesmarvin8** [comment]
Outside of removing the commented out code. I think just crediting portion of PSE code in a comment is all the remains.

Everything looks good.

File: halo2-shplonk/src/main.rs (hunk 2 above)
Comment (tyshko-rostyslav) [reply halo2-shplonk/src/main.rs]: Done

File: halo2-shplonk/src/main.rs (hunk 1 above)
Comment (tyshko-rostyslav) [reply halo2-shplonk/src/main.rs]: Done

**Review by jonesmarvin8 — APPROVED** [review]
Looks good.

**Pull request merged**